### ISSUE TITLE
Restructure rules bug

### ISSUE THREAD

**Issue by vijayaraghavanramanan** [jakubpawlowicz/clean-css#1031]
### Environment

Using the online tool
### Configuration options
Level 2 optimizations with merge semantically, restructure rules, remove unused @ rules chosen.

### Input CSS

```
.archive-header {
	margin-bottom: 24px;
	margin-bottom: 1.714285714rem;
}

img.aligncenter,
.wp-caption.aligncenter {
	clear: both;
	margin-top: 12px;
	margin-top: 0.857142857rem;
	margin-bottom: 12px;
	margin-bottom: 0.857142857rem;
}

.wp-caption.alignleft,
.wp-caption.alignright,
.wp-caption.aligncenter {
	margin-bottom: 24px;
	margin-bottom: 1.714285714rem;
}
```

### Actual output CSS (unminified for clarity)

```
.archive-header,.wp-caption.aligncenter,.wp-caption.alignleft,.wp-caption.alignright {
	margin-bottom: 24px;
	margin-bottom: 1.714285714rem
}

.wp-caption.aligncenter,img.aligncenter {
	clear: both;
	margin-top: 12px;
	margin-top: .857142857rem;
	margin-bottom: 12px;
	margin-bottom: .857142857rem
}
```

### Expected output CSS

.wp-caption.aligncenter should have a margin-bottom of 24px.

**Comment by tandrewnichols** [comment]
I'm having a similar problem with css arrows. I'm using this module via `less-plugin-clean-css` with `{ advanced: true }` (it works as expected with `false`).

I have this (simplified) less code using a mixin to generate the arrow css:

```
.arrow-box(@offset; @background; @position: bottom; @size: 8; @border: #e0e0e0; @border-width: 1; @positionOffsetDirection: left) {
  background: @background;
  border: unit(@border-width, px) solid @border;
  &:after, &:before {
    @{position}: 100%;
    border: solid transparent;
    content: " ";
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
    @{positionOffsetDirection}: @offset;
  }
  &:after {
    border-color: rgba(red(@background), green(@background), blue(@background), 0);
    border-@{position}-color: @background;
    border-width: unit(@size + @border-width, px);
    margin-@{positionOffsetDirection}: unit(-@size - @border-width, px);
    z-index: 100;
  }
  &:before {
    border-color: rgba(red(@border), green(@border), blue(@border), 0);
    border-@{position}-color: @border;
    border-width: unit(@size + @border-width*2, px);
    margin-@{positionOffsetDirection}: unit(-@size - @border-width*2, px);
    z-index: 100;
  }
}

#product-arrow {
  .arrow-box(193px, #fff);
}

#resource-arrow {
  .arrow-box(293px, #fff);
}
```

which produces the following css (unminified for clarity):

```
#product-arrow:after,
#product-arrow:before,
#resource-arrow:after,
#resource-arrow:before {
    bottom: 100%;
    content: " ";
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none
}

#product-arrow,
#resource-arrow {
    background: #fff;
    border: 1px solid #e0e0e0
}

#product-arrow:after,
#product-arrow:before {
    border: solid transparent;
    left: 193px
}

#product-arrow:after,
#resource-arrow:after {
    border-color: rgba(255, 255, 255, 0);
    border-bottom-color: #fff;
    border-width: 9px;
    margin-left: -9px;
    z-index: 100
}

#product-arrow:before,
#resource-arrow:before {
    border-color: rgba(224, 224, 224, 0);
    border-bottom-color: #e0e0e0;
    border-width: 10px;
    margin-left: -10px;
    z-index: 100
}

#resource-arrow:after,
#resource-arrow:before {
    border: solid transparent;
    left: 293px
}
```

Because it places `#resource-arrow:after,#resource-arrow:before` _last_, the border declaration for that ends up being transparent and therefore the arrow doesn't show for that arrow. The `#product-arrow` works fine because it's `border: solid transparent` is placed higher (above the `border-bottom-color` override).

**Comment by jakubpawlowicz** [comment]
I have verified this bug in the latest version on main branch and all works fine now.

Nonetheless I'm sorry about such a delay. Clean-css v5 #1135 will be out soon with many fixes to the latest 4.2.

**Comment by jakubpawlowicz** [comment]
Oh, see af64452559c2f717a61a5424b605d7b6906338a2 for your test cases included in the codebase. Thanks!
